src/main/java/com/rcv/ResultsWriter.java

```diff
@@ -543,23 +597,45 @@ void generateCdfJson(List<CastVoteRecord> castVoteRecords) throws IOException {
       for (int round = 1; round <= numRounds; round++) {
         List<Pair<String, BigDecimal>> currentRoundSnapshotData =
             cvr.getCdfSnapshotData().get(round);
+
+        // this is the hack
+        if(currentRoundSnapshotData == null && previousRoundSnapshotData == null) {
+          currentRoundSnapshotData = new ArrayList<>();
+          Pair<String, BigDecimal> placeholder = new Pair<>("Jacob Frey", BigDecimal.ONE);
+          currentRoundSnapshotData.add(placeholder);
+        }
+        // end hack
+
         if (currentRoundSnapshotData == null) {
           assert previousRoundSnapshotData != null; // this would indicate a bug in the tabulation
           currentRoundSnapshotData = previousRoundSnapshotData;
         }
         cvrSnapshots.add(generateCvrSnapshotMap(cvr, round, currentRoundSnapshotData));
         previousRoundSnapshotData = currentRoundSnapshotData;
       }
+      // create new cvr map entry
 
-      cvrMaps.add(
-          Map.ofEntries(
+      // if using precincts add GpUnitId for cvr precinct
+      if (config.isTabulateByPrecinctEnabled()) {
```

Comment:
I'm guessing you did this because `Map.ofEntries` returns an immutable Map. Let's not duplicate all of this logic, though. The cleanest way I've identified to handle a case like this is to forgo `ofEntries`. Instead, just create a normal empty Map, add each of the common entries using `put`, and then use an `if` block for your conditional `put`.